### DISCUSSION TITLE
Does Otterscan also include the $USD value of transaction at the time of a transaction?

### DISCUSSION THREAD

**Discussion by Etherdrake** [otterscan/otterscan#483] [General]
I was wondering if it is possible to calculate the $ value transacted of low-cap tokens with 100% accuracy and add this to the Otterscan.

**Comment by wmitsuda** [comment]
Do you have any suggestion about the methodology?

I mean, we currently display USD amounts from chainlink feeds at the time of tx (thanks historical states). It would be easy to add other converters given we know where/how to calculate this info from.

**Comment by Etherdrake** [comment]
I am working on an API for it myself, or at least have been theorizing it. Sadly, there is no source for the price of lower-cap tokens, and I was wondering if you guys found a way. I tried to find a way for 3 months and could not figure it out with 100% accuracy, although I did figure out how to calculate the correct price with about 99% of the values being accurate on certain tokens (basically the method involved a lot of extra code for every extra token you wanted to calculate the price for). 

Could you elaborate on the way you do it with Chainlink? I did not know they had historical prices, or do you mean the historical states of the oracles? It is true that this would only work for larger tokens that have a connection to Chainlink right?

**Comment by wmitsuda** [comment]
Yeah, Chainlink has an on chain feed for certain pairs, so we query for the feed existence and get the price. And since it is on chain and Erigon is an archive node, you get historical prices for free.

I actually do have some ideas for other price feeds, I just didn't implement them. Like, almost every token has an uniswap LP, so it should be able to estimate a spot price by doing the path token -> ETH on uniswap, then ETH -> USD using Chainlink.

That should give you a spot price, which should be different from the actual price if you executed a swap, depending on how deep is the token liquidity. But I think that scenario should be also possible to determine.

**Comment by sealer3** [comment]
We'll have spot price support for tokens pooled with ETH on Uniswap in https://github.com/otterscan/otterscan/pull/1899. The accuracy depends on the liquidity available in the pools, and for Uniswap V3, also depends on whether the actual token price is within the range of a liquidity position.

**Comment by sealer3** [comment]
Note that this has been implemented in [v2.5.0](https://github.com/otterscan/otterscan/releases/tag/v2.5.0). See the section about token price oracles in the docs if you are using a chain other than Ethereum mainnet and would like to configure this: https://docs.otterscan.io/config/options/oracles

**Comment by Nika250** [comment]
thanks for the info!
